dspace-api/src/main/java/org/dspace/content/RelationshipServiceImpl.java

```diff
@@ -378,50 +389,71 @@ private boolean checkMinCardinality(Context context, Item item,
 
     public List<Relationship> findByItemAndRelationshipType(Context context, Item item,
                                                             RelationshipType relationshipType, boolean isLeft)
-
         throws SQLException {
-        List<Relationship> list = this.findByItem(context, item);
-        List<Relationship> listToReturn = new LinkedList<>();
-        for (Relationship relationship : list) {
-            if (isLeft) {
-                if (StringUtils
-                    .equals(
-                        relationship.getRelationshipType().getLeftwardType(), relationshipType.getLeftwardType())
-                ) {
-                    listToReturn.add(relationship);
-                }
-            } else {
-                if (StringUtils
-                    .equals(
-                        relationship.getRelationshipType().getRightwardType(), relationshipType.getRightwardType())
-                ) {
-                    listToReturn.add(relationship);
-                }
-            }
-        }
-        return listToReturn;
+        List<Relationship> list = this.findByItemAndRelationshipType(context, item, relationshipType, -1, -1, isLeft);
+        return list;
     }
 
     @Override
     public List<Relationship> findByItemAndRelationshipType(Context context, Item item,
                                                             RelationshipType relationshipType)
-
         throws SQLException {
-        List<Relationship> list = this.findByItem(context, item);
-        List<Relationship> listToReturn = new LinkedList<>();
-        for (Relationship relationship : list) {
-            if (relationship.getRelationshipType().equals(relationshipType)) {
-                listToReturn.add(relationship);
-            }
-        }
-        return listToReturn;
+        List<Relationship> list = relationshipDAO.
+                findByItemAndRelationshipType(context, relationshipType, -1, -1, item);
+        return list;
+    }
+
+    @Override
+    public List<Relationship> findByItemAndRelationshipType(Context context, Item item,
+                                                            RelationshipType relationshipType, int limit, int offset)
+            throws SQLException {
+        List<Relationship> list = relationshipDAO.
+                findByItemAndRelationshipType(context, relationshipType, limit, offset, item);
+        return list;
+    }
+
+    @Override
+    public List<Relationship> findByItemAndRelationshipType(Context context, Item item,
+                                                            RelationshipType relationshipType, int limit, int offset,
+                                                            boolean isLeft)
```

Comment:
This should be:
```java
public List<Relationship> findByItemAndRelationshipType(Context context, Item item, RelationshipType relationshipType, boolean isLeft, int limit, int offset)

Reply:
Corrected: 
https://github.com/DSpace/DSpace/pull/2535/commits/240a142d9b9ae7e5fc4a22b5ad92637497b5ea11#diff-2e89fe18482e7ac43a1a4e901a8105b7R415